unattended.sh

```diff
@@ -1,186 +1,776 @@
-  systemctl enable httpd24-httpd mariadb rh-php73-php-fpm snmpd snmptrapd gorgoned centreontrapd cbd centengine centreon
-  systemctl restart httpd24-httpd mariadb rh-php73-php-fpm snmpd snmptrapd
-  print_step_end
-else
-  print_step_end "OK, systemd not detected, skipping"
-fi
+	log "INFO" "Centreon $topology installation from ${CENTREON_REPO}"
+
+	#FIXME : repo testing enabled for master
+	$PKG_MGR -q clean all --enablerepo="*" && $PKG_MGR -q install -y centreon --enablerepo="$CENTREON_REPO"
+
+	if [ "x$?" '!=' x0 ]; then
+		error_and_exit "Could not install Centreon (package centreon)"
+	fi
+
+	#
+	# PHP
+	#
+
+	log "INFO" "PHP configuration"
+	timezone=$($PHP_BIN -r '
+		$timezoneName = timezone_name_from_abbr(trim(shell_exec("date \"+%Z\"")));
+		if (preg_match("/Time zone: (\S+)/", shell_exec("timedatectl"), $matches)) {
+			$timezoneName = $matches[1];
+		}
+		if (date_default_timezone_set($timezoneName) === false) {
+			$timezoneName = "UTC";
+		}
+		echo $timezoneName;
+	' 2>/dev/null)
+	echo "date.timezone = $timezone" >$PHP_ETC/50-centreon.ini
+
+	log "INFO" "PHP date.timezone set to $timezone"
+
+	secure_mariadb_setup
+}
+#========= end of function install_central()
+
+#========= begin of function install_poller()
+# install the Centreon Poller
+#
+function install_poller() {
+	log "INFO" "Poller installation from ${CENTREON_REPO}"
+	$PKG_MGR -q clean all --enablerepo="*" && $PKG_MGR -q install -y centreon-poller-centreon-engine --enablerepo=$CENTREON_REPO
+	if [ "x$?" '!=' x0 ]; then
+		error_and_exit "Could not install Centreon (package centreon)"
+	fi
+}
+#========= end of function install_poller()
 
+#========= begin of function update_after_installation()
+# execute some tasks after having installed Centreon
+# - update firewall config
+# - enable some newly added services
 #
-# SUMMARY
+# ## FIXME -- according to the $topology
 #
+function update_after_installation() {
+
+	update_firewall_config
+
+	enable_new_services
+
+	set_runtime_selinux_mode $selinux_mode
+
+	set_selinux_config $selinux_mode
+
+}
+#========= end of function update_after_installation()
+
+#####################################################
+################ MAIN SCRIPT EXECUTION ##############
+
+## Process the provided arguments in line
+case "$1" in
+
+upgrade)
+	operation="upgrade"
+	parse_subcommand_options "$@"
+	;;
+
+install)
+	operation="install"
+	parse_subcommand_options "$@"
+	;;
+
+*)
+	log "WARN" "No provided operation : default value '$operation' will be used"
+	#usage
+	operation="install"
+	parse_subcommand_options "$@"
+	;;
+
+esac
+
+## Display all configured parameters
+log "INFO" "Start to execute operation [$operation] with following configuration parameters:"
+log "INFO" " topology   : \t$topology"
+log "INFO" " version    : \t$version"
+log "INFO" " repository : \t$repo"
+
+log "WARN" "It will start in '$default_timeout_in_sec' seconds. If you don't want to way, press any key to continue or Ctrl-C to exit"
```

Comment:
```suggestion
log "WARN" "It will start in '$default_timeout_in_sec' seconds. If you don't want to wait, press any key to continue or Ctrl-C to exit"
```